From dfefc6299937bba7a3256643c1f08aebcb2a0de9 Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Tue, 10 Feb 2015 13:08:49 +0800
Subject: [PATCH] usb/gadget: fix MaxBurst issue in FFS gadget

Since config_ep_by_speed() is not called from FFS gadget, it passes
the default endpoint.maxburst value 0 to device driver. This will
lead to wrong setting of DEPCFG's Parameter 0 as the minimum value
should be 1.

Change-Id: I3172d517722469021feef1710a51ebdf583309d1
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-16503
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
---
 drivers/usb/gadget/f_fs.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/gadget/f_fs.c b/drivers/usb/gadget/f_fs.c
index 3bb4d44..c090d06 100644
--- a/drivers/usb/gadget/f_fs.c
+++ b/drivers/usb/gadget/f_fs.c
@@ -1359,6 +1359,11 @@ static int ffs_func_eps_enable(struct ffs_function *func)
 
 		ep->ep->driver_data = ep;
 		ep->ep->desc = ds;
+
+		ret = config_ep_by_speed(ffs->gadget, &func->function, ep->ep);
+		if (ret)
+			break;
+
 		ret = usb_ep_enable(ep->ep);
 		if (likely(!ret)) {
 			epfile->ep = ep;
-- 
1.9.1

